pkg/controllers/secretstore/secretstore_controller.go

```diff
@@ -71,12 +86,183 @@ func (r *StoreReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	})
 }
 
+// handleSecretStoreFinalizer manages the finalizer for SecretStores
+// It adds a finalizer when there are PushSecrets with DeletionPolicy=Delete that reference this store
+// and removes it when there are no such PushSecrets
+func (r *StoreReconciler) handleSecretStoreFinalizer(ctx context.Context, ss *esapi.SecretStore, log logr.Logger) error {
```

Comment:
this logic should be unified with ClusterSecretStore imo, you can setup a. joint interface or directly write some helpers

Reply:
We have an interface in place already for this - the GenericStore. It could be used to add a SetFinalizer() / UnsetFinalizer() GetFinalizers() methods